influxdb_client/client/query_api.py

```diff
@@ -146,18 +179,34 @@ def query_data_frame_stream(self, query: str, org=None, data_frame_index: List[s
         if org is None:
             org = self._influxdb_client.org
 
-        response = self._query_api.post_query(org=org, query=self._create_query(query, self.default_dialect, params),
-                                              async_req=False, _preload_content=False, _return_http_data_only=False)
+        response = self._query_api.post_query(
+            org=org,
+            query=self._create_query(query, self.default_dialect, params, self._query_options),
+            async_req=False,
+            _preload_content=False,
+            _return_http_data_only=False)
+
+        _parser = FluxCsvParser(
+            response=response,
+            serialization_mode=FluxSerializationMode.dataFrame,
+            data_frame_index=data_frame_index,
+            query_options=self._query_options)
 
-        _parser = FluxCsvParser(response=response, serialization_mode=FluxSerializationMode.dataFrame,
-                                data_frame_index=data_frame_index)
         return _parser.generator()
 
-    # private helper for c
     @staticmethod
-    def _create_query(query, dialect=default_dialect, params: dict = None):
-        created = Query(query=query, dialect=dialect, extern=QueryApi._build_flux_ast(params))
-        return created
+    def _create_query(query, dialect=default_dialect, params: dict = None, query_options: QueryOptions = None):
+        q = Query(query=query, dialect=dialect, extern=QueryApi._build_flux_ast(params, query_options.profilers))
+
+        if query_options is not None \
+                and query_options.profilers is not None \
+                and len(query_options.profilers) > 0:
```

Comment:
Use `query_options` from `self` instead passing it as argument. It will be easier for future maintenance.

```suggestion
    def _create_query(self, query, dialect=default_dialect, params: dict = None):
        q = Query(query=query, dialect=dialect, extern=QueryApi._build_flux_ast(params, self._query_options.profilers))

        if self._query_options is not None \
                and self._query_options.profilers is not None \
                and len(self._query_options.profilers) > 0:
```